File: videogames-system/aws-resources/cdk/bin/cdk-app.ts
import 'source-map-support/register';
import type { App, Environment } from 'aws-cdk-lib';
import { BasicStack } from '../lib/basic-stack.stack';

const buildStack = (app: App): App => {
  const BaseName = 'Basic-Account-Infrastructure';
  const region = 'eu-west-1';

  const stackEnv: Environment = {
    region,
    account: process.env.CDK_DEFAULT_ACCOUNT
  };

  new BasicStack(app, BaseName, {
    env: stackEnv,
    stackBaseName: BaseName
  });

  return app;
};

export default buildStack;

File: videogames_system/aws_resources/cdk/bin/cdk_app.py
from aws_cdk import App, Environment

from videogames_system.aws_resources.cdk.lib.main_stack import MainStack, StackBaseProps


def build_stack(app: App) -> App:
    """
    Builds and returns a stack for the given app.

    Parameters:
        app (App): The app instance.

    Returns:
        App: The app instance with the stack built.
    """
    base_name = "Basic-CDK-Infrastructure"
    region = "eu-west-1"

    stack_env = Environment(region=region, account=app.account)
    stack_props = StackBaseProps(stack_base_name=base_name, env=stack_env)

    MainStack(app, base_name, props=stack_props)

    return app
